xclim/temperature.py

```diff
@@ -75,33 +75,114 @@ def cfprobe(self, dan, dax):
+                keywords='',
+                compute=_ind.tn_max,
+                )
 
-cold_spell_duration = Tasmin(identifier='cold_spell_duration',
+tn_min = Tasmin(identifier='tnmin',
+                long_name='Minimum daily minimum temperature',
+                units='K',
+                standard_name='air_temperature',
+                cell_methods='time: minimum within days time: minimum over days',
+                description='{freq} minimum of daily minimum temperature.',
+                keywords='',
+                compute=_ind.tn_min,
+                )
+
+daily_temperature_range = TasminTasmax(identifier='dtr',
+                                       long_name='Mean Diurnal Temperature Range',
+                                       units='K',
+                                       standard_name='air_temperature',
+                                       cell_methods='time range within days time: mean over days',
+                                       description='{freq} mean diurnal temparature range',
+                                       compute=_ind.daily_temperature_range,
+                                       )
+
+daily_temperature_range_variability = TasminTasmax(identifier='dtrvar',
+                                                   long_name='Mean Diurnal Temperature Range Variability',
+                                                   units='K',
+                                                   standard_name='air_temperature',
+                                                   cell_methods='time range within days time: difference '
+                                                                'over days time: mean over days',
+                                                   description='{freq} mean diurnal temparature range variability ('
+                                                               'defined as the average day-to-day variation '
+                                                               'in daily temperature range '
+                                                               'for the given time period)',
+                                                   compute=_ind.daily_temperature_range_variability,
+                                                   )
+
+extreme_temperature_range = TasminTasmax(identifier='etr',
+                                         long_name='Intra-period Extreme Temperature Range',
+                                         units='K',
+                                         standard_name='air_temperature',
+                                         description='{freq} range between the maximum of max temperature (TXx)'
+                                                     ' and the minimum of min temperature (TNn)',
+                                         compute=_ind.extreme_temperature_range,
+                                         )
+
+cold_spell_duration = Tasmin(identifier='csdi{window}',
```

Comment:
csdi_{window} ?

Reply:
Underscores for both cases of {thresh} and {window} would be my preference but am trying to fit with the metadata sheet as much as possible... Also noticed that we may want to ensure that {thresh} or {window} values are converted to integers before being string formatted (e.g. some functions have 'thresh=5.0' as defaults ,,, gives 'gdd5.0'..  Can we think of examples where thresh would actually have a decimal?

Reply:
Also, For the specific case of csdi (as well as some others) metadata doesn't incorporate the window size (e.g. ICCLIM is always 6).